lighthouse-core/report/html/renderer/pwa-category-renderer.js

```diff
@@ -46,6 +46,42 @@ class PwaCategoryRenderer extends CategoryRenderer {
     return categoryElem;
   }
 
+  /**
+   * @param {LH.ReportResult.Category} category
+   * @return {DocumentFragment}
+   */
+  renderScoreGauge(category) {
+    // Defer to parent-gauge style if category error.
+    if (category.score === null) {
+      return super.renderScoreGauge(category);
+    }
+
+    const tmpl = this.dom.cloneTemplate('#tmpl-lh-gauge--pwa', this.templateContext);
+    const wrapper = /** @type {HTMLAnchorElement} */ (this.dom.find('.lh-gauge--pwa__wrapper',
+      tmpl));
+    wrapper.href = `#${category.id}`;
+
+    const passingGroupIds = this._getPassingGroupIds(category.auditRefs);
+    let className = 'lh-badged-na';
+    if (passingGroupIds.has('pwa-fast-reliable')) {
```

Comment:
😆

Reply:
to clarify the laughing was on this whole nesting business we need to do, not just this line :)

FWIW, I'd probably find something like 

```js
const isReliable = passingGroupIds.has()
const isInstallable = ...
const isOptimized = ...

if (isReliable && isInstallable && isOptimized) className = 'lh-badged-all'
else if (isReliable && isInstallable) className = 'lh-badged-reliable-installable'
...
```

a bit easier to read, but no biggie

Reply:
ha, that is significantly easier to read. I was very tired when I wrote all that is my only excuse :)

While you were writing this I was just working on a more css-centric version. What do you think of this instead?

The looooong class names are the only downside. Maybe we can short circuit the BEMness to pick short names